Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Upsert.ts

```diff
@@ -4,26 +4,43 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 import type { PoolClient, QueryResult } from 'pg';
-import { UpsertResult, UpsertRequest, Logger } from '@edfi/meadowlark-core';
-import { getDocumentInsertOrUpdateSql, getRecordExistsSql } from './QueryHelper';
+import {
+  UpsertResult,
+  UpsertRequest,
+  Logger,
+  DocumentReference,
+  documentIdForDocumentReference,
+} from '@edfi/meadowlark-core';
+
+import {
+  getDeleteReferencesSql,
+  getDocumentInsertOrUpdateSql,
+  getCheckDocumentExistsSql,
+  getReferencesInsertSql,
+} from './QueryHelper';
+import { validateReferences } from './WriteHelper';
 
 export async function upsertDocument(
   { id, resourceInfo, documentInfo, edfiDoc, validate, traceId, security }: UpsertRequest,
   client: PoolClient,
 ): Promise<UpsertResult> {
-  const upsertResult: UpsertResult = { response: 'UNKNOWN_FAILURE' };
+  let upsertResult: UpsertResult = { response: 'UNKNOWN_FAILURE' };
 
   let recordExistsResult: QueryResult;
-  // let outRefs;
-  let documentSql: string;
+  let documentUpsertSql: string;
   let isInsert: boolean;
 
+  const outRefs = documentInfo.documentReferences.map((dr: DocumentReference) => documentIdForDocumentReference(dr));
+
   try {
-    recordExistsResult = await client.query(getRecordExistsSql(id));
+    recordExistsResult = await client.query(getCheckDocumentExistsSql(id));
 
-    isInsert = !recordExistsResult.rowCount || recordExistsResult.rowCount === 0;
+    isInsert = !recordExistsResult.rows[0].exists;
```

Comment:
There's 0.0% chance that `rows == null`?

Reply:
I believe so, since the query is now a sub select it should return true or false no matter what, but will research a bit to verify

Reply:
I would assume that the PostgreSQL driver might react in some strange way and return nothing. Because if we don't assume that, and it does happen, then we end up with a nasty error that is harder to debug.

Reply:
Discussed with Stephen re: hardening error detection overall for PostgeSQL will move forward as is and revisit in RND-299